File: app/local_plugins/core/ai_plugin_interface/android/src/main/kotlin/com/aiagent/plugin_interface/AgentEventSink.kt
package com.aiagent.plugin_interface

/**
 * Agent 事件回调接口
 *
 * 由 agents_server 的 AgentsServerPlugin 实现，
 * 接收来自各 Agent 类型插件的事件，转发给 Flutter EventChannel。
 */
interface AgentEventSink {
    fun onSttEvent(event: SttEventData)
    fun onLlmEvent(event: LlmEventData)
    fun onTtsEvent(event: TtsEventData)
    fun onStateChanged(sessionId: String, state: String, requestId: String?)
    fun onError(sessionId: String, errorCode: String, message: String, requestId: String?)

    /** 端到端连接状态变化（disconnected / connecting / connected / error） */
    fun onConnectionStateChanged(sessionId: String, state: String, errorMessage: String? = null)

    /**
     * Agent 就绪回调（统一替代"等 connected"的协议）。
     *
     * 每次 [NativeAgent.connectService] 调用后，agent 必须最终派发恰好一次：
     *  - ready=true  : 三段式 = 服务初始化完成；端到端 = 链路 connected
     *  - ready=false : errorCode/errorMessage 必填，编排器据此终止流程
     *
     * disconnect/release 后再次 connect 才会重新派发。
     */
    fun onAgentReady(
        sessionId: String,
        ready: Boolean,
        errorCode: String? = null,
        errorMessage: String? = null,
    )
}

File: app/local_plugins/core/ai_plugin_interface/ios/Classes/AgentEventSink.swift
import Foundation

/// Agent → AgentsServer fan-out for all per-session events.
///
/// Implemented by `AgentsServerPlugin` (which forwards to a Flutter
/// EventChannel). Each registered `NativeAgent` is handed an instance and
/// pushes STT / LLM / TTS / state / error / connection events through it.
public protocol AgentEventSink: AnyObject {
    func onSttEvent(_ event: SttEventData)
    func onLlmEvent(_ event: LlmEventData)
    func onTtsEvent(_ event: TtsEventData)
    func onStateChanged(sessionId: String, state: String, requestId: String?)
    func onError(sessionId: String, errorCode: String, message: String, requestId: String?)
    func onConnectionStateChanged(sessionId: String, state: String, errorMessage: String?)

    /// Agent-ready callback (replaces the older "wait for connected" pattern).
    /// Exactly one ready-event must fire per `connectService` call, with
    /// `ready=true` on success or `ready=false` plus error info on failure.
    func onAgentReady(sessionId: String, ready: Bool, errorCode: String?, errorMessage: String?)
}

public extension AgentEventSink {
    func onConnectionStateChanged(sessionId: String, state: String, errorMessage: String? = nil) {
        onConnectionStateChanged(sessionId: sessionId, state: state, errorMessage: errorMessage)
    }
    func onAgentReady(sessionId: String, ready: Bool) {
        onAgentReady(sessionId: sessionId, ready: ready, errorCode: nil, errorMessage: nil)
    }
}

/// Service-test event fan-out (mirrors `AgentEventSink` but for service_manager).
public protocol ServiceTestEventSink: AnyObject {
    func onSttTestEvent(testId: String, kind: String, text: String?, errorCode: String?, errorMessage: String?)
    func onTtsTestEvent(testId: String, kind: String, progressMs: Int?, durationMs: Int?, errorCode: String?, errorMessage: String?)
    func onLlmTestEvent(
        testId: String,
        kind: String,
        textDelta: String?,
        thinkingDelta: String?,
        toolCallId: String?,
        toolName: String?,
        toolArgumentsDelta: String?,
        toolResult: String?,
        fullText: String?,
        errorCode: String?,
        errorMessage: String?
    )
    func onTranslationTestEvent(
        testId: String,
        kind: String,
        sourceText: String?,
        translatedText: String?,
        sourceLanguage: String?,
        targetLanguage: String?,
        errorCode: String?,
        errorMessage: String?
    )
    func onStsTestEvent(testId: String, kind: String, text: String?, state: String?, errorCode: String?, errorMessage: String?)
    func onAstTestEvent(testId: String, kind: String, text: String?, state: String?, errorCode: String?, errorMessage: String?)
    func onTestDone(testId: String, success: Bool, message: String?)
}
